### ISSUE TITLE
Critical Process

### ISSUE THREAD

**Issue by gamethrower** [UnamSanctam/SilentXMRMiner#449]
What is the downsides of using "BSoD if killed"? Would be nice to have this in miner.

**Comment by UnamSanctam** [comment]
Well, means you can't uninstall it mainly.

**Comment by gamethrower** [comment]
Is this the only reason?

**Comment by UnamSanctam** [comment]
Problem is also that it will BSoD if the program crashes or just exits normally as well, if the program closes in any way it will BSoD.

**Comment by gamethrower** [comment]
Even when shutting down the PC?

**Comment by UnamSanctam** [comment]
Don't think so, but if you input the wrong connection settings (or lose internet connection) in the miner then it will eventually close (tries for 5 minutes I think or something), and then you would BSoD since it will close down.

**Comment by gamethrower** [comment]
Yeah, now I see that this would be a little bit rude to have such feature :rofl:

**Comment by UnamSanctam** [comment]
Haha yeah, I did experiment a bit with making the process a debugging process which means you can't close the process easily, I did have some success but have to test more. I'll also be releasing a new version of the XMR and ETH miners today or tomorrow mainly focused on decreasing detections.

**Comment by gamethrower** [comment]
How do I modify the code to make the miner start only after reboot?

**Comment by gamethrower** [comment]
Can you explain in detail what this xmrig arguments do?
 `--cpu-memory-pool=1 --randomx-no-rdmsr`

**Comment by UnamSanctam** [comment]
In the new version  of XMRig (that my miner also uses).

**Comment by gamethrower** [comment]
Is this the "ok" amount of shares rejected with "Block expired" reason or I should reduce latency? 
![image](https://user-images.githubusercontent.com/46888050/137926985-ffcb469c-4891-4b6e-99a1-2a52dc6515f1.png)

**Comment by UnamSanctam** [comment]
Not really, it's such a small amount that it doesn't really matter but if you still want to lower it then you can just change pool until you find one which works better.

**Comment by gamethrower** [comment]
When I test miner on my PC (3 threads) with 70% CPU load it uses 1 threads instead of 2. Before I launch it CPU usage doesn't go above 10%, so I guess 2 threads should be available. Why it uses 33% instead of 66%?

**Comment by gamethrower** [comment]
How can I check what cores are idle? Win7 task manager doesn't show that. Here's what it shows when I start the miner:
![image](https://user-images.githubusercontent.com/46888050/137944620-aa2d3af6-42e3-4ed9-aeaa-af2768ee674a.png)

**Comment by UnamSanctam** [comment]
If you mean the normal xmrig then it will say whether it set the MSR registers correctly or if it didn't have enough permissions.

**Comment by gamethrower** [comment]
It doesn't say anything about MSR even if I delete the sys file.

**Comment by UnamSanctam** [comment]
Take a screenshot.

**Comment by UnamSanctam** [comment]
Yes that's what I do in my miner.

**Comment by gamethrower** [comment]
I have checked the default limit and it's 1024. Doesn't it mean 1024 constant connections?

**Comment by UnamSanctam** [comment]
All hard-forks that change the algorithm do that, since they change the algorithm, unless they hard-fork to an algorithm that the miner already has.

**Comment by UnamSanctam** [comment]
But it doesn't really matter if you have "Remote Configuration" since you can just mine any of the hundreds of other cryptocurrencies supported by the miner.

**Comment by UnamSanctam** [comment]
Yes as long as the algorithm stays it will work, and will also work if they ever change it if the algorithm already exists in my miner. But yes I do not think there is any reason to change it at the moment, even more cryptocurrencies have also started using variations of RandomX as well.
